src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -37,13 +37,15 @@ void CoreWorkerDirectTaskSubmitter::AddWorkerLeaseClient(
 }
 
 void CoreWorkerDirectTaskSubmitter::OnWorkerIdle(
-    const rpc::WorkerAddress &addr, const SchedulingKey &scheduling_key, bool was_error,
+    const rpc::WorkerAddress &addr, const SchedulingKey &scheduling_key,
+    bool is_actor_creation, bool was_error,
```

Comment:
Can we handle the actor creation elsewhere instead of in this function? It is quite large.

Also, rescheduling for the worker is logically separate from registering actors.